tests/gui/models/selection-state.spec.js

```diff
@@ -512,6 +512,40 @@ describe('Browser: SelectionState', function() {
           }).to.throw('Invalid image archive extension: 1');
         });
 
+        it('should throw if the archive extension doesn\'t match the last path extension', function() {
+          m.chai.expect(function() {
+            SelectionStateModel.setImage({
+              path: 'foo.img.xz',
+              extension: 'img',
+              archiveExtension: 'gz',
+              size: {
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Image archive extension mismatch: gz and xz');
+        });
+
+        it('should throw if the extension is not recognised', function() {
+          m.chai.expect(function() {
+            SelectionStateModel.setImage({
+              path: 'foo.img.xz',
+              extension: 'ifg',
+              archiveExtension: 'xz',
+              size: {
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Invalid image extension: ifg');
+        });
+
```

Comment:
From what I can tell of https://github.com/resin-io/etcher/pull/1347/files#diff-74bcbd72e100aecd2f68e833e7a169f4R342 this is a newly-caught error? Bonus!
Could we also add a test for `{ path: 'foo.ifg', extension: 'ifg' }` ?  (and perhaps `{ path: 'foo.ifg.gz', extension: 'ifg', archiveExtension: 'gz' }` too?)